### PR TITLE
fix(alerts): Add filters to incidents for alert rule details

### PR DESCRIPTION
**Pull request by taylangocmen**
This adds start/end and detailed parameters to the incidents endpoint for the alert rule details page. Also added `groupStatsPeriod: 'auto' ` to related issues and changed the breadcrumbs in alert rule details.

### CODE REVIEW

File: src/sentry/incidents/endpoints/organization_incident_index.py [hunk 1]
```diff
@@ -34,6 +35,14 @@ def get(self, request, organization):
...
             incidents = incidents.filter(alert_rule=query_alert_rule)
 
+        query_start = request.GET.get("start")
+        if query_start is not None:
+            incidents = incidents.filter(start=query_start)
+
+        query_end = request.GET.get("end")
+        if query_end is not None:
+            incidents = incidents.filter(end=query_end)
+
```
Comment (wedamija) [inline src/sentry/incidents/endpoints/organization_incident_index.py:45]: You need tests to cover this.

`incidents.filter(start__gte=query_start)`, or `__gt` depending on whether you want `>=` or `>`.

File: src/sentry/incidents/endpoints/organization_incident_index.py [hunk 2]
```diff
@@ -49,11 +58,19 @@ def get(self, request, organization):
             # Filter to only error alerts
             incidents = incidents.filter(alert_rule__snuba_query__dataset=Dataset.Events.value)
 
+        query_detailed = request.GET.get("detailed")
+
+        def serialize_results(results):
+            if query_detailed:
+                return serialize(results, request.user, DetailedIncidentSerializer())
+            else:
+                return (serialize(results, request.user),)
```
Comment (wedamija) [inline src/sentry/incidents/endpoints/organization_incident_index.py:67]: Is this meant to be a tuple?

File: src/sentry/static/sentry/app/views/alerts/rules/details/relatedIssues.tsx [hunk 3]
```diff
@@ -66,10 +65,10 @@ class RelatedIssues extends React.Component<Props> {
             orgId={organization.slug}
             endpointPath={path}
             queryParams={queryParams}
-            query=""
+            query={`start=${start}&end=${end}&groupStatsPeriod=auto`}
```
Comment (davidenwang) [inline src/sentry/static/sentry/app/views/alerts/rules/details/relatedIssues.tsx:68]: just curious whats the difference between queryParams and query here? seems like they have a lot of the same info

**Review by davidenwang — APPROVED** [review]
FE looks good

File: src/sentry/incidents/endpoints/organization_incident_index.py (hunk 2 above)
Comment (taylangocmen) [reply src/sentry/incidents/endpoints/organization_incident_index.py]: It wasn't

File: src/sentry/static/sentry/app/views/alerts/rules/details/relatedIssues.tsx (hunk 3 above)
Comment (taylangocmen) [reply src/sentry/static/sentry/app/views/alerts/rules/details/relatedIssues.tsx:68]: Honestly they should be reduced into 1, but as of now we use `queryParams` for issues requests and query to populate the global selection (as well as some parameters for a few requests like `groupStatsPeriod`, not a global selection but still works the same way)

File: src/sentry/incidents/endpoints/organization_incident_index.py [hunk 4]
```diff
@@ -34,6 +35,14 @@ def get(self, request, organization):
         if query_alert_rule is not None:
             incidents = incidents.filter(alert_rule=query_alert_rule)
 
+        query_start = request.GET.get("start")
+        if query_start is not None:
+            incidents = incidents.filter(start__gte=query_start)
+
+        query_end = request.GET.get("end")
+        if query_end is not None:
+            incidents = incidents.filter(end__gte=query_end)
```
Comment (wedamija) [inline src/sentry/incidents/endpoints/organization_incident_index.py:44]: We still need tests to make sure this works

Comment (taylangocmen) [reply src/sentry/incidents/endpoints/organization_incident_index.py]: Added them

**Review by wedamija — APPROVED** [review]

**Pull request merged**